File: samples/basic-samples/simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/screens/HomeKey.java
package com.zhuinden.navigationexamplefrag.screens;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestackextensions.fragments.DefaultFragmentKey;

import androidx.fragment.app.Fragment;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class HomeKey extends DefaultFragmentKey {
    public static HomeKey create() {
        return new AutoValue_HomeKey();
    }

    @Override
    protected Fragment instantiateFragment() {
        return new HomeFragment();
    }
}

File: samples/basic-samples/simple-stack-example-basic-kotlin-fragment/src/main/java/com/zhuinden/simplestackexamplekotlinfragment/screens/HomeKey.kt
package com.zhuinden.simplestackexamplekotlinfragment.screens

import com.zhuinden.simplestackextensions.fragments.DefaultFragmentKey
import kotlinx.parcelize.Parcelize

/**
 * Created by Owner on 2017.11.13.
 */
@Parcelize
data object HomeKey : DefaultFragmentKey() { // generate reliable `toString()` for no-args data class
    override fun instantiateFragment() = HomeFragment()
}
